Modules/mathmodule.c

```diff
@@ -2195,12 +2195,10 @@ math_modf_impl(PyObject *module, double x)
     double y;
     /* some platforms don't do the right thing for NaNs and
        infinities, so we take care of special cases directly. */
-    if (!Py_IS_FINITE(x)) {
-        if (Py_IS_INFINITY(x))
-            return Py_BuildValue("(dd)", copysign(0., x), x);
-        else if (Py_IS_NAN(x))
-            return Py_BuildValue("(dd)", x, x);
-    }
+    if (Py_IS_INFINITY(x))
+        return Py_BuildValue("(dd)", copysign(0., x), x);
+    else if (Py_IS_NAN(x))
+        return Py_BuildValue("(dd)", x, x);
```

Comment:
Indeed, this is less efficient for ``Py_IS_FINITE()`` case.  But I think we could rewrite this with ``else`` clause instead of ``else if (Py_IS_NAN(x))``.

Reply:
Maybe. If neither of them has bugs.

Reply:
I think we could be safe that no libm is so broken in 2023.  Also it's already assumed in many places across the module (e.g. in ``m_tgamma()``, ``m_lgamma()``, ``m_log2()``, etc).

Reply:
Thanks.  Patch added to #110000.